google-cloud-pubsub/src/main/java/com/google/cloud/pubsub/v1/MessageDispatcher.java

```diff
@@ -361,7 +391,8 @@ void processReceivedMessages(List<ReceivedMessage> messages) {
       AckRequestData ackRequestData = builder.build();
       AckHandler ackHandler =
           new AckHandler(ackRequestData, message.getMessage().getSerializedSize(), totalExpiration);
-      if (pendingMessages.putIfAbsent(message.getAckId(), ackHandler) != null) {
+      if (!this.exactlyOnceDeliveryEnabled.get()
```

Comment:
nit. We are checking twice if exactlyOnceDelivery is enabled. Let's just remove the check here, and move this whole block into the else statement below for exactly once delivery not enabled. This will help a lot with code clarity